src/lib/stepper/step/_core.scss

```diff
@@ -0,0 +1,377 @@
+
+  cursor: not-allowed;
+}
+
+@mixin title-disabled {
+  color: #{token(disabled-text-color)};
+}
+
+@mixin subtitle-disabled {
+  color: #{token(disabled-text-color)};
+}
+
+@mixin forge-interactable() {
+  cursor: pointer;
+}
+
+@mixin icon-content-default-theme {
+  border-color: #{token(primary-color)};
+}
+
+@mixin icon-content-default-theme-index {
+  color: #{token(primary-color)};
+}
+
+@mixin icon-content() {
+  color: #{token(icon-text-color)};
+  background-color: #{token(icon-fill)};
+  height: #{token(icon-content-size)};
+  width: #{token(icon-content-size)};
+
+  border-style: solid;
+  border-width: 2px;
+  border-color: transparent;
+  border-radius: 50%;
+
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  transition: background-color 400ms cubic-bezier(0.25, 0.8, 0.25, 1);
+  margin: 0 8px 0 0;
+  flex: none;
+
+  forge-icon {
+    font-size: #{token(icon-size)};
+  }
+}
+
+@mixin icon-content-selected() {
+  @include override(icon-fill, icon-fill-active);
+  @include override(icon-text-color, icon-text-color-active);
+}
+
+@mixin icon-content-completed() {
+  @include icon-content-selected();
+}
+
+@mixin icon-content-error {
+  @include override(icon-fill, error-color);
+  @include override(icon-text-color, error-text-color);
+}
+
+@mixin icon-content-disabled {
+  @include override(icon-fill, disabled-color);
+  @include override(icon-text-color, disabled-text-color);
+}
+
+@mixin line() {
+  border-color: #{token(line-color)};
+
+  border-top-style: solid;
+  border-top-width: 1px;
+  border-radius: 1px 0 0 1px;
+}
+
+@mixin step-line() {
+  border-color: #{token(line-color)};
+  min-width: #{token(line-min-width)};
+
+  content: '';
+  position: relative;
+  height: 0;
+  flex: 1;
+  flex-basis: 0.000000001px;
+  border-top-style: solid;
+  border-top-width: 1px;
+  border-radius: 1px 0 0 1px;
+}
+
+@mixin text-container() {
+  display: flex;
+  flex-direction: column;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+@mixin text-container-alternate {
+  margin-top: 8px;
+  place-items: center;
+}
+
+@mixin label {
+  @include typography.style(body1);
+
+  color: #{token(label-color)};
+
+  text-align: left;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+@mixin label-selected {
+  color: #{token(primary-color)};
+
+  font-weight: 500;
+}
+
+@mixin label-error {
+  color: #{token(error-color)};
+}
+
+@mixin optional-label {
+  color: #{token(sub-label-color)};
+
+  @include typography.style(label1);
+}
+
+@mixin optional-label-slot-wrapper {
+  text-align: left;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+@mixin optional-label-selected {
+  color: #{token(primary-color)};
+
+  font-weight: 500;
+}
+
+@mixin optional-label-error {
+  color: #{token(error-color)};
+}
+
+@mixin line-clustered {
+  @include override(line-min-width, line-min-width-clustered);
+}
+
+@mixin vertical-base {
+  width: 100%;
+  border-radius: 4px;
```

Comment:
We need to propagate this `border-radius` to the focus indicator when it's vertical. Once the base `border-radius` is using a shape token, you can just override it here instead and then it would automatically propagate to the focus indicator.